cf_xarray/accessor.py

```diff
@@ -214,7 +214,7 @@ def _get_custom_criteria(
                 if re.match(patterns, obj[var].attrs.get(criterion, "")):
                     results.update((var,))
                 # also check name specifically since not in attributes
-                elif criterion == "name" and re.match(patterns, var):
+                elif criterion == "name" and re.match(patterns, str(var)):
```

Comment:
isn't this wrong when `var` is a Hashable like a tuple?

We could add a `and not isinstance(var, ReprObject)` instead. I think it's ok to depend on that private class

Reply:
maybe `and isinstance(var, str)` for now.

Reply:
Sure that works for me